### ISSUE TITLE
Error in plugin.py on Windows with Python 3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8610]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)| @rodrigc reported|
|-|-|
|Trac ID|trac#8610|
|Type|enhancement|
|Created|2016-07-17 00:50:18Z|
|Branch|https://github.com/twisted/twisted/tree/8610-rodrigc-filepath-py3|

I did the following on Windows with Python 3.5:

`tox -e py35-tests-windows`

and got this error in plugin.py

```
py35-tests-windows runtests: PYTHONHASHSEED='217'
py35-tests-windows runtests: commands[0] | C:\projects\twisted\build\py35-tests-windows\Scripts/trial --reactor=default --reporter=verbose twisted
Unexpected error while writing cache file
Traceback (most recent call last):
  File "c:\projects\twisted\build\py35-tests-windows\lib\site-packages\twisted\python\usage.py", line 411, in <lambda>
    fn = lambda name, value, m=method: m(value)
  File "c:\projects\twisted\build\py35-tests-windows\lib\site-packages\twisted\application\app.py", line 517, in opt_reactor
    installReactor(shortName)
  File "c:\projects\twisted\build\py35-tests-windows\lib\site-packages\twisted\application\reactors.py", line 80, in installReactor
    for installer in getReactorTypes():
  File "c:\projects\twisted\build\py35-tests-windows\lib\site-packages\twisted\plugin.py", line 213, in getPlugins
    allDropins = getCache(package)
--- <exception caught here> ---
  File "c:\projects\twisted\build\py35-tests-windows\lib\site-packages\twisted\plugin.py", line 185, in getCache
    dropinPath.setContent(pickle.dumps(dropinDotCache))
  File "c:\projects\twisted\build\py35-tests-windows\lib\site-packages\twisted\python\filepath.py", line 1530, in setContent
    os.rename(sib.path, self.path)
builtins.ValueError: rename: src and dst must be the same type
Unexpected error while writing cache file
Traceback (most recent call last):
  File "c:\projects\twisted\build\py35-tests-windows\lib\site-packages\twisted\scripts\trial.py", line 472, in postOptions
    _BasicOptions.postOptions(self)
  File "c:\projects\twisted\build\py35-tests-windows\lib\site-packages\twisted\scripts\trial.py", line 382, in postOptions
    self['reporter'] = self._loadReporterByName(self['reporter'])
  File "c:\projects\twisted\build\py35-tests-windows\lib\site-packages\twisted\scripts\trial.py", line 369, in _loadReporterByName
    for p in plugin.getPlugins(itrial.IReporter):
  File "c:\projects\twisted\build\py35-tests-windows\lib\site-packages\twisted\plugin.py", line 213, in getPlugins
    allDropins = getCache(package)
--- <exception caught here> ---
  File "c:\projects\twisted\build\py35-tests-windows\lib\site-packages\twisted\plugin.py", line 185, in getCache
    dropinPath.setContent(pickle.dumps(dropinDotCache))
  File "c:\projects\twisted\build\py35-tests-windows\lib\site-packages\twisted\python\filepath.py", line 1530, in setContent
    os.rename(sib.path, self.path)
builtins.ValueError: rename: src and dst must be the same type

```

<details><summary>Searchable metadata</summary>

```
trac-id__8610 8610
type__enhancement enhancement
reporter__rodrigc rodrigc
priority__normal normal
milestone__Python_3_x Python-3.x
branch__8610_rodrigc_filepath_py3 8610-rodrigc-filepath-py3
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1468716618559704 1468716618559704
changetime__1470047133496526 1470047133496526
version__None None
owner__rodrigc rodrigc

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_2"></a>|@rodrigc commented|
|-|-|

https://github.com/twisted/twisted/pull/449

**Comment by twisted-trac** [comment]
|[<img alt="Lukasa's avatar" src="https://avatars.githubusercontent.com/u/1382556?s=50" width="50" height="50">](https://github.com/Lukasa)<a name="note_3"></a>|@Lukasa set owner to @rodrigc|
|-|-|

Ok, this looks fine to me. As always, I'm nervous that we haven't obviously enabled any tests, or in any way demonstrated that this actually changes behaviour in our tests, but given the lack of a Windows Python 3 test bot there's not much we can do.

One note that's optional for you to deal with, otherwise feel free to merge.

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_4"></a>|GitHub set status to `closed`|
|-|-|

In changeset 6d9cd4bc1ca1b1aed07de4b8e3166c44885931da
```
#!CommitTicketReference repository="" revision="6d9cd4bc1ca1b1aed07de4b8e3166c44885931da"
Merge pull request #449 from twisted/8610-rodrigc-filepath-py3

Author: hawkowl, rodrigc
Reviewer: Lukasa, glyph
Fixes: #8610

Fix twisted.python.filepath.setContent() on Windows with Python 3
```
